rules/terraformrules/terraform_naming_convention.go

```diff
@@ -77,91 +99,68 @@ func (r *TerraformNamingConventionRule) Check(runner *tflint.Runner) error {
 		return fmt.Errorf("Invalid default configuration: %v", err)
 	}
 
-	if err := r.checkDataBlocks(runner, &config, defaultNameValidator); err != nil {
-		return err
+	var dataTargets, localTargets, moduleTargets, outputTargets, resourceTargets, variableTargets []RuleTarget
```

Comment:
Avoid collecting data into new slices like this. Iterate directly over the maps that Terraform has already allocated for each block type. 👍 to eliminating `r.checkLocalValues` and similar heavily duplicated methods, but it should be possible to do so without copying data ahead of time.

Reply:
I made the corrections to iterate the maps directly.

Reply:
@bendrucker I made the corrections to iterate the maps directly. Hope that's better now.